src/Controller/SitemapController.php

```diff
@@ -4,31 +4,28 @@
 
 namespace SitemapPlugin\Controller;
 
-use SitemapPlugin\Builder\SitemapBuilderInterface;
-use SitemapPlugin\Renderer\SitemapRendererInterface;
-use Symfony\Component\HttpFoundation\Request;
+use SitemapPlugin\Filesystem\Reader;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
 use Symfony\Component\HttpFoundation\Response;
 
 final class SitemapController extends AbstractController
 {
-    /** @var SitemapBuilderInterface */
-    protected $sitemapBuilder;
+    /** @var ChannelContextInterface */
+    private $channelContext;
 
     public function __construct(
-        SitemapRendererInterface $sitemapRenderer,
-        SitemapBuilderInterface $sitemapBuilder
+        ChannelContextInterface $channelContext,
+        Reader $reader
     ) {
-        $this->sitemapRenderer = $sitemapRenderer;
-        $this->sitemapBuilder = $sitemapBuilder;
+        $this->channelContext = $channelContext;
+
+        parent::__construct($reader);
     }
 
-    public function showAction(Request $request): Response
+    public function showAction(string $name): Response
     {
-        $filter = [];
-        if ($request->attributes->has('name')) {
-            $filter[] = $request->attributes->get('name');
-        }
+        $path = \sprintf('%s/%s', $this->channelContext->getChannel()->getCode(), \sprintf('%s.xml', $name));
```

Comment:
I use the [Safe library](https://github.com/thecodingmachine/safe) for functions like this. That way I don't have to check for `sprintf` returning false for example.

Reply:
Looks interesting, let's look into that later. I think we are currently good, as the things we are doing are rather safe or else it will break the whole installation anyway (if there is no channel e.g.).